src/otx/algo/detection/rtdetr.py

```diff
@@ -0,0 +1,438 @@
+        self.optimizer_configuration = optimizer_configuration
+
+    def _forward_features(self, images: Tensor, targets: dict[str, Any] | None = None) -> dict[str, Tensor]:
+        images = self.backbone(images)
+        images = self.encoder(images)
+        return self.decoder(images, targets)
+
+    def forward(self, images: Tensor, targets: dict[str, Any] | None = None) -> dict[str, Tensor] | Tensor:
+        """Forward pass of the model."""
+        original_size = images.shape[-2:]
+
+        if self.multi_scale and self.training:
+            sz = int(np.random.choice(self.multi_scale))
+            images = nn.functional.interpolate(images, size=[sz, sz])
+
+        output = self._forward_features(images, targets)
+        if self.training:
+            return self.criterion(output, targets)
+        return self.postprocess(output, original_size)
+
+    def postprocess(
+        self,
+        outputs: dict[str, Tensor],
+        original_size: tuple[int, int] | None = None,
+        deploy_mode: bool = False,
+    ) -> dict[str, Tensor] | tuple[list[Tensor], list[Tensor], list[Tensor]]:
+        """Post-processes the model outputs.
+
+        Args:
+            outputs (dict[str, Tensor]): The model outputs.
+            original_size (tuple[int, int], optional): The original size of the input images. Defaults to None.
+            deploy_mode (bool, optional): Whether to run in deploy mode. Defaults to False.
+
+        Returns:
+            dict[str, Tensor] | tuple[list[Tensor], list[Tensor], list[Tensor]]: The post-processed outputs.
+        """
+        logits, boxes = outputs["pred_logits"], outputs["pred_boxes"]
+
+        # convert bbox to xyxy and rescale back to original size (resize in OTX)
+        bbox_pred = torchvision.ops.box_convert(boxes, in_fmt="cxcywh", out_fmt="xyxy")
+        if not deploy_mode and original_size is not None:
+            original_size_tensor = torch.tensor(original_size).to(bbox_pred.device)
+            bbox_pred *= original_size_tensor.repeat(1, 2).unsqueeze(1)
+
+        # perform scores computation and gather topk results
+        scores = nn.functional.sigmoid(logits)
+        scores, index = torch.topk(scores.flatten(1), self.num_top_queries, axis=-1)
+        labels = index % self.num_classes
+        index = index // self.num_classes
+        boxes = bbox_pred.gather(dim=1, index=index.unsqueeze(-1).repeat(1, 1, bbox_pred.shape[-1]))
+
+        if deploy_mode:
+            return {"bboxes": boxes, "labels": labels, "scores": scores}
+
+        scores_list, boxes_list, labels_list = [], [], []
+
+        for sc, bb, ll in zip(scores, boxes, labels):
+            scores_list.append(sc)
+            boxes_list.append(
+                torchvision.tv_tensors.BoundingBoxes(bb, format="xyxy", canvas_size=original_size),
+            )
+            labels_list.append(ll.long())
+
+        return scores_list, boxes_list, labels_list
+
+    def export(
```

Comment:
It's not really related to this PR, but it doesn't seem common for `nn.Module` to have a function called `export`, and most of the detection models in OTX currently have a similar structure. Is this something that needs to be refactored? Or is it OTX's logic to add an `export` function to `nn.Module` when adding a custom model (nn.Module)?

Reply:
I think yes, I would rather have common interface for all our tasks. Now, we have difference on many parts. Concerning export -> we need to choose the proper method for that. "export", "deploy" or just "forward_for_tracing". For now, I just added it in the same way as for all object detection models. But I will keep it in mind for the refactoring task in OTX2.2